tests/serialization/test_filter.py

```diff
@@ -92,20 +96,76 @@ def test_raise_on_type(value, filter_value, op):
 
 
 @pytest.mark.parametrize("op", ["==", "!=", ">=", "<=", ">", "<"])
-@pytest.mark.parametrize("col", list("ABCDE"))
-def test_filter_df_from_predicates(op, col):
+@pytest.mark.parametrize(
+    "data,value",
+    [
+        (
+            # data
+            range(10),
+            # value
+            4,
+        ),
+        (
+            # data
+            ["A", "B"] * 5,
+            # value
+            "A",
+        ),
+        (
+            # data
+            pd.Series(["X", "Y"] * 5).astype("category"),
+            # value
+            "X",
+        ),
+        (
+            # data
+            pd.Series([datetime.date(2019, 1, 1), datetime.date(2019, 1, 2)] * 5),
+            # value
+            datetime.date(2019, 1, 1),
+        ),
+        (
+            # data
+            [datetime.datetime(2019, 1, 1), datetime.datetime(2019, 1, 2)] * 5,
+            # value
+            datetime.datetime(2019, 1, 1),
+        ),
+        (
+            # data
+            [datetime.datetime(2019, 1, 1), datetime.datetime(2019, 1, 2)] * 5,
+            # value
+            datetime.date(2019, 1, 1),
+        ),
+        (
+            # data
+            np.arange(10, dtype=np.uint8),
+            # value
+            4,
+        ),
+    ],
+)
+def test_filter_df_from_predicates(op, data, value):
+    df = pd.DataFrame({"A": data})
+    df["B"] = range(len(df))
+
+    predicates = [[("A", op, value)]]
+    actual = filter_df_from_predicates(df, predicates)
+    if pd.api.types.is_categorical(df["A"]):
+        df["A"] = df["A"].astype(df["A"].cat.as_ordered().dtype)
+    if isinstance(value, datetime.date) and (df["A"].dtype == "datetime64[ns]"):
+        # silence pandas warning
+        value = pd.Timestamp(value)
```

Comment:
If it's about silencing a warning, use filter warnings, e.g. https://docs.pytest.org/en/latest/warnings.html#pytest-mark-filterwarnings

This casting otherwise defeats the purpose of the test, doesn't it?

Reply:
pandas will reject data<->datetime comparisons entirely in the future. We can either silence the warning and break soon or can decide to always be strict and stop all this magic altogether or support it and therefore need a slightly more complicates test.

Reply:
Yes, it is supposed to raise. I would've expected that *we* take care of the appropriate casting already or is it about the expected value?

Reply:
ok, my bad... I should start reading the code more carefully... sorry